File: web/data-structures/AVLNode.php
<?php 

/**
* AVL Node model
*/
class AVLNode {
  public $data;
  public $parent;
  public $left;
  public $right;
  public $balance = 0;


  function __construct($parent, $data, $left, $right) {
    $this->parent = $parent;
    $this->data = $data;
    $this->left = $left;
    $this->right = $right;
  }

  // $node = AVLNode::withParentAndChildren($parent, $data, $left, $right);
  static function withParentAndChildren($parent, $data, $left, $right) {
    return new self($parent, $data, $left, $right);
  }

  static function withParent($parent, $data) {
    return new self($parent, $data, null, null);
  }

  static function withData($data) {
    return new self(null, $data, null, null);
  }


}




 ?>

File: android/AVL/AVLNode.java
package com.example.rolando.myapplication_buenaversiob.AVL;

/**
 * Created by Rolando on 10/11/2016.
 */
public class AVLNode<E> {
    E data;
    AVLNode parent = null;
    AVLNode left = null;
    AVLNode right = null;
    Integer balance = 0;
    AVLNode(E data){
        this(null,data,null,null);
    }
    AVLNode(AVLNode parent, E data){
        this(parent,data,null,null);
    }
    AVLNode(AVLNode parent, E data, AVLNode left, AVLNode right){
        this.parent = parent;
        this.data = data;
        this.left = left;
        this.right = right;
    }

    public E getData() {
        return data;
    }

    public void setData(E data) {
        this.data = data;
    }

    public AVLNode getParent() {
        return parent;
    }

    public void setParent(AVLNode parent) {
        this.parent = parent;
    }

    public AVLNode getLeft() {
        return left;
    }

    public void setLeft(AVLNode left) {
        this.left = left;
    }

    public AVLNode getRight() {
        return right;
    }

    public void setRight(AVLNode right) {
        this.right = right;
    }

    public Integer getBalance() {
        return balance;
    }

    public void setBalance(Integer balance) {
        this.balance = balance;
    }


}
